PORTAL/jobs/_pyperformance.py

```diff
@@ -1816,24 +2071,29 @@ def _match_filename(self, filename):
         )
         return filename == self.filename
 
-    def find_comparison(self, comparisons):
+    def find_comparison(
+            self,
+            comparisons: PyperfComparisons
+    ) -> Optional[PyperfComparison]:
         try:
             return comparisons.bysource[self.filename]
         except KeyError:
             return None
 
-    def load_results(self):
+    def load_results(self) -> Optional["PyperfResults"]:
         resfile = self.resfile
         if not resfile:
             return None
         return resfile.read()
 
-    def as_rendered_row(self, columns):
-        results = None
+    def as_rendered_row(self, columns: Iterable[str]) -> List[str]:
         row = []
         for column in columns:
             if column == 'date':
-                rendered = self.date.strftime('%Y-%m-%d (%H:%M UTC)')
+                if self.date is None:
+                    rendered = "(unknown)"
```

Comment:
Ha!  I wasn't expecting the date to ever be `None`, but apparently it can happen if `self.filename` isn't set.  Go figure.